### PR TITLE
E2E tests: use built artefacts in tests

### PR DESCRIPTION
**Pull request by adimoldovan**
#### Changes proposed in this Pull Request:

Add another e2e tests run that uses build workflow artefacts.
Currently restricted to trunk and release branches. 
After Build workflow completes successfully, the e2e workflow will download the build artefacts, extract them, and will use the tested plugin artefact in the e2e tests instead of running the build command.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
pd5faL-gm-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* This won't run until merged to trunk.
* To test locally:
	* Download the uploaded artefacts of a build workflow and extract them in the monorepo root
	* Run `.github/files/e2e-tests/prepare-plugins-for-e2e-env.sh` script. Check e2e workflow for the required env variables.
	* In the e2e project run `pnpm env:up` and then check the plugin

### CODE REVIEW

File: .github/files/e2e-tests/map-plugins-for-e2e-env.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	echo "::error::PROJECT_PATH must be set"
+	exit 1
+fi
+
+tar --xz -xvvf "$BUILD_DIR/jetpack-build/build.tar.xz" -C "$BUILD_DIR"
+
+SLUG=$(jq -r -e ".ci.pluginSlug" "$PROJECT_PATH/package.json")
+MIRROR=$(jq -r -e ".ci.mirrorName" "$PROJECT_PATH/package.json")
+
+echo "$MIRROR"
```
Comment (brbrr) [inline .github/files/e2e-tests/map-plugins-for-e2e-env.sh]: Looks like a debugging leftover. Do we need this?

**Review by brbrr — DISMISSED** [review]
LGTM

File: .github/files/e2e-tests/map-plugins-for-e2e-env.sh (hunk 1 above)
Comment (adimoldovan) [reply .github/files/e2e-tests/map-plugins-for-e2e-env.sh]: Leftover indeed. It's not needed.

Comment (adimoldovan) [reply .github/files/e2e-tests/map-plugins-for-e2e-env.sh]: Removed with 616a661

**Review by brbrr — APPROVED** [review]

**Pull request merged**